### PR TITLE
Gh 245 proxy store issue

### PR DESCRIPTION
**Pull request by go159lab**
fix federated store schema issues and code refactoring

### CODE REVIEW

File: gaffer-as-a-service/gaas-rest/src/main/resources/config/federated.yaml [hunk 1]
```diff
@@ -1,11 +1,4 @@
 graph:
-  config:
-    hooks:
-      - auths:
-          uk.gov.gchq.gaffer.federatedstore.operation.AddGraph:
-            - GAAS_SYSTEM_USER
```
Comment (t92549) [inline gaffer-as-a-service/gaas-rest/src/main/resources/config/federated.yaml:6]: Isn't this removing the auth that only lets GAAS admin AddGraphs?

File: gaffer-as-a-service/gaas-rest/src/test/java/uk/gov/gchq/gaffer/gaas/factories/GafferFactoryTest.java [hunk 2]
```diff
@@ -143,310 +146,99 @@ void mapStoreStoreRequestShouldReturnMapStoreRequestBody() {
...
-                        "\"uk.gov.gchq.gaffer.federatedstore.FederatedStore\"},\"config\":{\"configName\":\"federatedBig\",\"description\":\"Another description\",\"graphId\":" +
-                        "\"MyGraph\",\"hooks\":[{\"class\":\"uk.gov.gchq.gaffer.graph.hook.OperationAuthoriser\",\"auths\":" +
-                        "{\"uk.gov.gchq.gaffer.federatedstore.operation.AddGraph\":[\"GAAS_SYSTEM_USER\"]}}]},\"operationDeclarations\":[{\"handler\":{\"class\":" +
-                        "\"uk.gov.gchq.gaffer.proxystore.operation.handler.GetProxyUrlHandler\"},\"operation\":\"uk.gov.gchq.gaffer.proxystore.operation.GetProxyUrl\"}]}," +
-                        "\"ingress\":{\"host\":\"mygraph-kai-dev.apps.my.kubernetes.cluster\",\"pathPrefix\":{\"ui\":\"/ui\",\"api\":\"/rest\"}}}}";
-        assertEquals(expected, gson.toJson(requestBody));
-    }
-
-    @Test
-    void federatedBigStoreRequestWithValidOpAuthsInTheConfigYamlWithoutAddGraphClass_shouldOverrideToOneConfiguredInGafferSpecConfig() {
```
Comment (t92549) [inline gaffer-as-a-service/gaas-rest/src/test/java/uk/gov/gchq/gaffer/gaas/factories/GafferFactoryTest.java:166]: Why are these tests removed?

File: gaffer-as-a-service/gaas-rest/src/main/java/uk/gov/gchq/gaffer/gaas/factories/GafferFactory.java [hunk 3]
```diff
@@ -101,89 +94,34 @@ private static GafferSpec overrideGafferSpecConfig(final GafferSpec config, fina
...
+        if (existingOperationDeclarations != null) {
+            List<OperationDeclaration> operations = existingOperationDeclarations.getOperations();
+            operations.forEach((operation) -> declarations.getOperations().add(operation));
         }
-        final Map<String, Object> proxyUrlDeclaration = new HashMap<>();
-        final Map<String, String> proxyUrlClass = new HashMap<>();
-
-        proxyUrlClass.put("class", "uk.gov.gchq.gaffer.proxystore.operation.handler.GetProxyUrlHandler");
 
-        proxyUrlDeclaration.put("operation", "uk.gov.gchq.gaffer.proxystore.operation.GetProxyUrl");
```
Comment (t92549) [inline gaffer-as-a-service/gaas-rest/src/main/java/uk/gov/gchq/gaffer/gaas/factories/GafferFactory.java:158]: Why is GetProxyUrl being removed?

Comment (go159lab) [reply gaffer-as-a-service/gaas-rest/src/main/java/uk/gov/gchq/gaffer/gaas/factories/GafferFactory.java:158]: Replaced with this code line: 108-113

  final OperationDeclarations declarations = new OperationDeclarations.Builder()
                .declaration(new OperationDeclaration.Builder()
                        .handler(new GetProxyUrlHandler())
                        .operation(GetProxyUrl.class)
                        .build())
                .build();

**Comment by go159lab** [comment]
@t92549 I will create a new ticket to address both your comments  (the auth that only lets GAAS admin AddGraphs? and related Junit tests )

File: gaffer-as-a-service/gaas-rest/src/test/java/uk/gov/gchq/gaffer/gaas/factories/GafferFactoryTest.java (hunk 2 above)
Comment (rocky341) [reply gaffer-as-a-service/gaas-rest/src/test/java/uk/gov/gchq/gaffer/gaas/factories/GafferFactoryTest.java:166]: I think that they are related to the auth hooks

**Review by t92549 — APPROVED** [review]

**Comment by rocky341** [comment]
@t92549 
Ready to merge when you are

**Pull request merged**